assets/bigquery/loitering-events.sql.j2

```diff
@@ -79,28 +79,22 @@ SELECT
     lat_max,
     lon_max,
     regions AS regions_mean_position,
-    convert_m_to_km((
-        SELECT distance_from_shore_m
-        FROM `{{ spatial_measures_table }}`
-        WHERE gridcode = format_gridcode(lon_min, lat_min)
-    )) AS start_distance_from_shore_km,
-    convert_m_to_km((
-        SELECT distance_from_shore_m
-        FROM `{{ spatial_measures_table }}`
-        WHERE gridcode = format_gridcode(lon_min, lat_min)
+    convert_m_to_km(
+      spatial_measures_start.distance_from_shore_m
+    ) AS start_distance_from_shore_km,
+    convert_m_to_km(
+      spatial_measures_end.distance_from_shore_m
     )) AS end_distance_from_shore_km,
-    convert_m_to_km((
-        SELECT distance_from_port_m
-        FROM `{{ spatial_measures_table }}`
-        WHERE gridcode = format_gridcode(lon_max, lat_max)
-    )) AS start_distance_from_port_km,
-    convert_m_to_km((
-        SELECT distance_from_port_m
-        FROM `{{ spatial_measures_table }}`
-        WHERE gridcode = format_gridcode(lon_max,lat_max)
-    )) AS end_distance_from_port_km,
+    convert_m_to_km(
+      spatial_measures_start.distance_from_port_m
+    ) AS start_distance_from_port_km,
+    convert_m_to_km(
+      spatial_measures_end.distance_from_port_m
+    ) AS end_distance_from_port_km,
     event_info,
     event_vessels
 FROM base_loitering_event
-    JOIN `{{ spatial_measures_table }}` sm ON
-        format_gridcode(lon_mean, lat_mean) = sm.gridcode;
\ No newline at end of file
+JOIN `{{ spatial_measures_table }}` spatial_measures_start ON
+    format_gridcode(lon_min, lat_min) = sm.gridcode;
```

Comment:
Are we sure we're always going to find a record in `{{ spatial_measures_table }}` with `gridcode = format_gridcode(lon_min, lat_min)`? If it's not found we are going to miss those events. If we should include always the events then we should use `LEFT JOIN` here.

Also note that the alias is `spatial_measures_start` so the condition should be 
```suggestion
    format_gridcode(lon_min, lat_min) = spatial_measures_start.gridcode
```
I also removed the trailing `;`


The same applies for `spatial_measures_end`.

Reply:
Good catch on the ambiguous field name!

Regarding the spatial measures thing, I'm not decided. That table is supposed to have a global raster, so every gridcode should be there, and if it's not then we have a worse problem than not including some events. I prefer losing events if the spatial measures table is wrong but not allow NULL distances or regions.

Reply:
Perfect! Sounds reasonable to not include events if the spatial measures table is wrong. But we could also miss some events if the boundaries are not calculated or with wrong coordinates or if the mean position is missing or wrong too. I don't know if we have sensors in the previous steps of the pipeline to ensure this does not happen, but if not it would be great to add some sensors to detect and report this inconsistencies as soon as possible.